rasa/cdu/command_processor.py

```diff
@@ -358,8 +373,29 @@ def clean_up_commands(
             structlogger.debug(
                 "command_executor.skip_command.slot_already_set", command=command
             )
+        elif isinstance(command, SetSlotCommand) and command.name not in slots_so_far:
+            use_slot_fill = all(
+                step.question != command.name or step.skip_if_filled
+                for flow in all_flows.underlying_flows
+                for step in flow.get_question_steps()
+            )
+
+            if use_slot_fill:
+                clean_commands.append(command)
+            else:
+                structlogger.debug(
+                    "command_executor.skip_command.slot_not_asked_for", command=command
```

Comment:
Shouldn't these branches be in reverse? Doesn't `step.skip_if_filled` is False and `step.question == command_name` mean that we should still fill the slot?

Reply:
yes and no 😂 so the branches are right but your hint that `use_slot_fill` wasn't set properly is a good one, there were two cases that wasn't properly covering :+1: